lib/hammer_cli/output/definition.rb

```diff
@@ -1,21 +1,68 @@
 module HammerCLI::Output
-
   class Definition
-
     attr_accessor :fields
 
     def initialize
       @fields = []
     end
 
-    def append(fields)
+    def append(fields = nil, &block)
+      fields = [fields].compact unless fields.is_a?(Array)
       @fields += fields
+      return @fields unless block_given?
+      dsl = Dsl.new
+      dsl.build(&block)
+      @fields += dsl.fields
+    end
+
+    def find_field(field_id)
+      @fields[field_index(field_id)]
+    end
+
+    def insert(mode, field_id, fields = nil, &block)
+      index = field_index(field_id)
+      index += 1 if mode == :after
+      definition = self.class.new
+      definition.append(fields, &block)
+      insert_fields(index, definition.fields, with_remove: mode == :replace)
+    end
+
+    def at(path = [])
+      path = [path] unless path.is_a? Array
+      return self if path.empty?
+
+      field = find_field(path[0])
+
+      unless field.respond_to?(:output_definition)
+        raise ArgumentError, "Field #{path[0]} doesn't have nested output definition"
+      end
+
+      field.output_definition.at(path[1..-1])
+    end
+
+    def clear
+      @fields = []
     end
 
     def empty?
       @fields.empty?
     end
 
-  end
+    private
 
+    def field_index(field_id)
+      index = @fields.find_index do |f|
+        f.id == field_id
```

Comment:
I had an idea to match the field by ANY of those identifiers like:
```ruby
f.match_id?(field_id)
```
where the matcher is defined as
```ruby
def match_id?(field_id)
  @options[:id] == filed_id || @options[:key] == field_id || @label == field_id
end
```
It could prevent breaking the extension when e.g. the field_id is added.
What do you think?